spec/support/mailer_helper.rb

```diff
@@ -15,26 +15,116 @@ def strip_tags(str)
     ActionController::Base.helpers.strip_tags(str)
   end
 
+  # @param [String,String[],nil] to The email address(es) the message must've been sent to.
+  # @param [String,nil] subject The subject the email must've had
+  # @param [String[],nil] Array of substrings that must appear in body.
   def expect_delivered_email(to: nil, subject: nil, body: nil)
+    email = find_sent_email(to:, subject:, body:)
+
+    error_message = <<~ERROR
+      Unable to find email matching args:
+        to: #{to}
+        subject: #{subject}
+        body: #{body}
+      Sent mails:
+      #{summarize_all_deliveries(to:, subject:, body:).indent(2)}
+    ERROR
+
+    expect(email).to_not be(nil), error_message
+  end
+
+  # @param [String,nil] to If provided, the email address the message must've been sent to
+  # @param [String,nil] subject If provided, the subject the email must've had
+  # @param [String[],nil] Array of substrings that must appear in body.
+  def expect_email_not_delivered(to: nil, subject: nil, body: nil)
     email = ActionMailer::Base.deliveries.find do |sent_mail|
-      next unless to.present? && sent_mail.to == to
-      next unless subject.present? && sent_mail.subject == subject
-      if body.present?
+      to_matches = to.nil? || sent_mail.to.include?(to)
+      subject_matches = subject.nil? || sent_mail.subject == subject
+      body_matches = body.nil? || begin
         delivered_body = sent_mail.text_part.decoded.squish
-        body.to_a.each do |expected_body|
-          next unless delivered_body.include?(expected_body)
+        body.to_a.all? do |expected_substring|
+          delivered_body.include?(expected_substring)
         end
       end
-      true
+
+      to_matches && subject_matches && body_matches
     end
 
     error_message = <<~ERROR
-      Unable to find email matching args:
+      Found an email matching the below (but shouldn't have):
         to: #{to}
         subject: #{subject}
         body: #{body}
       Sent mails: #{ActionMailer::Base.deliveries}
     ERROR
-    expect(email).to_not be(nil), error_message
+
+    expect(email).to be(nil), error_message
+  end
+
+  private
+
+  def body_matches(email:, body:)
+    return true if body.nil?
+
+    delivered_body = email.text_part.decoded.squish
+
+    body.to_a.all? do |expected_substring|
+      delivered_body.include?(expected_substring)
+    end
+  end
+
+  def to_matches(email:, to:)
+    return true if to.nil?
+
+    to = Array.wrap(to).to_set
```

Comment:
I honestly had to read this a couple of times before I understood it.  I'm still not clear on exactly how this works when `to` is an array of strings.

Reply:
Sorry, I need to rebase a few things. This PR is actually a lot smaller.